src/ActionSheetProvider.tsx

```diff
@@ -1,41 +1,45 @@
 import * as React from 'react';
 
-import ActionSheet from './ActionSheet';
+// NativeActionSheet will always be custom when on Android/web
+import NativeActionSheet from './ActionSheet';
+import CustomActionSheet from './ActionSheet/CustomActionSheet';
 import { Provider } from './context';
-import { ActionSheetOptions, ActionSheetProps } from './types';
+import { ActionSheetOptions } from './types';
 
 interface Props {
   children: React.ReactNode;
   useNativeDriver?: boolean;
+  useCustomActionSheet?: boolean;
 }
 
-export default class ActionSheetProvider extends React.Component<Props> {
-  _actionSheetRef: React.RefObject<ActionSheet>;
-  _context: ActionSheetProps;
+export default function ActionSheetProvider({
+  children,
+  useNativeDriver,
+  useCustomActionSheet = false,
+}: Props) {
+  const actionSheetRef = React.useRef<NativeActionSheet>(null);
 
-  constructor(props: Props) {
-    super(props);
-    this._actionSheetRef = React.createRef();
-    this._context = {
+  const context = React.useMemo(
+    () => ({
       showActionSheetWithOptions: (options: ActionSheetOptions, callback: (i: number) => void) => {
-        if (this._actionSheetRef.current != null) {
-          this._actionSheetRef.current.showActionSheetWithOptions(options, callback);
+        if (actionSheetRef.current) {
+          actionSheetRef.current.showActionSheetWithOptions(options, callback);
         }
       },
-    };
-  }
+    }),
+    [actionSheetRef]
+  );
 
-  getContext(): ActionSheetProps {
-    return this._context;
-  }
+  const ActionSheet = React.useMemo(
```

Comment:
There's no need to memoize this, since it's a simple check that is returning a static variable in either case.

Reply:
It’s returning the same instance of the component function on every render, so I’m not exactly sure how memoizing could make a difference

Reply:
I made a reproduction here to show memorization does nothing, since you’re just assigning a variable to a static variable: https://codesandbox.io/s/eloquent-voice-yvnfn5?file=/src/App.js

Reply:
🤦🏻 you're right. Good catch!